platform/engineering/design-docs/2021-05-18-forms-library.md

```diff
@@ -0,0 +1,852 @@
+```
+
+**Discussion:** Is not ensuring the shape of the data matches the schema a
+reasonable boundary for the library to stop at?
+
+**Discussion:** What tools might Formulate give to developers to help catch
+these kinds of data shape mismatch mistakes?
+
+### Code Location
+Formulate will live in its own repository.
+
+### Documentation
+Formulate documentation will follow the [diataxis framework](https://diataxis.fr/).
+
+The following documentation is considered essential for the successful adoption
+of Formulate.
+- [Tutorials](https://diataxis.fr/tutorials/)
+  - Getting started
+  - Building a multi-page form
+- [How-to guides](https://diataxis.fr/how-to-guides/)
+  - How to build a review page
+  - How to make array pages
+  - How to make single-page list loops
+  - How to build a custom field
+  - How to write a data migration
+- [Reference documentation](https://diataxis.fr/reference/) tooling
+  - See[Reference documentation](#reference-documentation) below
+- [Explanations](https://diataxis.fr/explanation/)
+  - Why keep form state in a React context for multi-page forms?
+
+All documentation will be located within the Formulate repo in the `docs/`
+directory. It is essential that documentation and code live side-by-side so the
+documentation may be updated at the same time as the code. Additionally, keeping
+all the documentation in `docs/` allows us to serve it simply using GitHub Pages.
+
+#### Tutorials
+Tutorials will serve as the entry point for learning how to use Formulate.
+Because the level of effort maintaining tutorials is high, they will be
+intentionally few. Tutorials may be found in `docs/tutorials/`.
+
+#### How-to guides
+These guides will cover how to perform common tasks using Formulate. They will
+live in the `docs/how-to-guides/` directory of the repo.
+
+#### Reference documentation
+Automated API documentation will be generated by
+[`api-extractor`](https://api-extractor.com) in conjunction with
+`api-documenter` to create markdown files which will then be hosted on GitHub
+Pages, and linked to from both the [Documentation Front
+Page](https://department-of-veterans-affairs.github.io/va.gov-team/) and the
+[Documentation
+Site](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/).
+
+**Example:** Microsoft's own Rushstack—which is the monorepo where
+[`api-extractor`](https://github.com/microsoft/rushstack/tree/master/apps/api-extractor)
+is developed—uses Jekyll to create HTML pages from the markdown files generated
+by
+[`api-documenter`](https://github.com/microsoft/rushstack/tree/master/apps/api-documenter).
+See the
+[`rushstack.io-website`](https://github.com/microsoft/rushstack.io-website) repo
+and [corresponding website](https://rushstack.io/pages/api/).
+
+The built reference documentation will live in `docs/reference/`.
+
+#### Explanations
+Architecture decisions and other explanations for decisions made for the
+library will live in `docs/explanations/`.
```

Comment:
I'm assuming we will link back to this document as well?

Reply:
Yeah, that's a good idea! :raised_hands: